src/screens/Distribution/RingChart.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+
+import React, { PureComponent } from "react";
+import { PanResponder, View } from "react-native";
+import * as d3shape from "d3-shape";
+import Svg, { Path, G, Circle } from "react-native-svg";
+import colors from "../../colors";
+import type { DistributionItem } from "./DistributionCard";
+
+type Props = {
+  data: Array<DistributionItem>,
+  side: number,
+  style?: *,
+  selectedKey: number => void,
+  highlight?: number,
+};
+
+class RingChart extends PureComponent<Props> {
+  static defaultProps = {
+    data: [],
+  };
+
+  panResponder: *;
+  arcGenerator = d3shape.arc();
+  offsetX = 0;
+  offsetY = 0;
+  highlight = -1;
+  paths: * = {};
+
+  constructor(props: Props) {
+    super(props);
+    const { highlight, side, selectedKey, data } = this.props;
+
+    if (highlight) {
+      this.highlight = highlight;
+    }
+
+    this.panResponder = PanResponder.create({
+      onPanResponderGrant: event => {
+        this.offsetX = event.nativeEvent.pageX - event.nativeEvent.locationX;
+        this.offsetY = event.nativeEvent.pageY - event.nativeEvent.locationY;
+      },
+      onStartShouldSetPanResponder: () => true,
+      onStartShouldSetPanResponderCapture: () => false,
+      onMoveShouldSetPanResponder: () => false,
+      onMoveShouldSetPanResponderCapture: () => false,
+
+      onPanResponderMove: event => {
+        const x = event.nativeEvent.pageX - this.offsetX;
+        const y = event.nativeEvent.pageY - this.offsetY;
+
+        const y0 = side / 2;
+        const x0 = side / 2;
+        let rad = Math.atan2(y - y0, x - x0);
+        rad =
+          ((rad > 0 ? rad : 2 * Math.PI + rad) + 1.5707963268) % (2 * Math.PI);
```

Comment:
```suggestion
          (2.5 * Math.PI + rad) % (2 * Math.PI);
```

to confirm it works but i think it's equivalent code

Reply:
Seems to work 🤷‍♂